### PR TITLE
Add unmaintained `dotenv` & Co.

### LINKED ISSUE #1254: `dotenv` crate is implicitly unmaintained
As of May 21st, 2022, https://github.com/dotenv-rs/dotenv 's latest version is 0.15.0, which was published on October 22nd, 2019. And the latest commit is [3c1a77bc95821777e5ceb996c5e0b082f2a3ea38](https://github.com/dotenv-rs/dotenv/commit/3c1a77bc95821777e5ceb996c5e0b082f2a3ea38), which was pushed on Jun 27th, 2020.
On Dec 24th, 2021, someone asked the project status on [Current maintenance state · Issue #74 · dotenv-rs/dotenv](https://github.com/dotenv-rs/dotenv/issues/74) but there's no response from the maintainers.
I'm not sure how long "prolonged period" refers to, but this crate is a candidate for an "unmaintained" crate, I think. At least we should monitor how things are going there.

### PR DESCRIPTION
**Pull request by pinkforest**
Closes #1254 

@JohnTitor @est31 @8573 would you mind reviewing pls

Should we recommend dotenvy as an alternative or something else as per https://github.com/rustsec/advisory-db/issues/1254#issuecomment-1214313669 ?

Note: I would be slightly hesistant to proceed with unmaintained when the crate says it is intended to be used only in test / dev env - nonetheless there are associated monorepo dotenv_ crates which don't mention this and seem to be used elsewhere

EDIT: the associated monorepo crates also mention about intended use in dev / test only.

Thx.

### CODE REVIEW

**Comment by pinkforest** [comment]
(I would be slightly hesitant with this since it is meant for development / testing only - but there are associated crates which don't mention it)

**Comment by 8573** [comment]
I would not object either (1) to adding this advisory now or (2) to waiting until Dylan-DPC is added as an owner of `dotenvy` on Crates.io, so that the bus factor is raised.

File: crates/dotenv_codegen_implementation/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+```toml
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "dotenv_codegen_implementation"
```
[the remark this answers is not in the document]
Comment (pinkforest) [reply crates/dotenv_codegen_implementation/RUSTSEC-0000-0000.md]: No, that is correct good catch - it's redundant. removed :+1:

**Comment by pinkforest** [comment]
The crate implies and is pretty transparent that it is not to be used in production environment outside dev or test.

For this reason I am proposing to close this issue.

If someone has objection please this raise.

**Comment by allan2** [comment]
> Both crates imply - and are pretty transparent - that it is not to be used in production environment outside dev or test.

@pinkforest Hi, I am the maintainer of the `dotenvy` crate. Also responding to [your comment on #1254](https://github.com/rustsec/advisory-db/issues/1254#issuecomment-1214313669) here.

As stated on the README, dotenvy is convenient for dev environments. This does not mean that it is not intended for prod. 
Some may want to use *.env* files in dev only, preferring to set env vars in the VM or container in prod. Others may wish to use *.env* in both dev and prod environments. It's up to the preference of the developer.

Also noting that Dylan-DPC has been added to the repo on Aug 14.

Release v0.15.2 was also put out today. Thanks for your interest in dotenv/dotenvy!

**Comment by pinkforest** [comment]
Okay. Proposing to merge then and let's see if anyone objects.

**Comment by pinkforest** [comment]
Since the maintainers are truly not reachable and this is widely used, we'll merge and nobody is objecting.

Regardless that the crate says it is only for dev/test there is evidence that it is used more than that and we should perhaps advise on that.

**Review by amousset — APPROVED** [review]
Given there is a maintained fork this is easily actionable and I agree we should publish these.

**Pull request merged**